sympy/combinatorics/subsets.py

```diff
@@ -59,11 +58,20 @@ def __new__(cls, subset, superset):
             if elem not in superset:
                 raise ValueError('The superset provided is invalid as it does '
                                  'not contain the element {}'.format(elem))
-        obj = Basic.__new__(cls)
+        obj = object.__new__(cls)
         obj._subset = subset
         obj._superset = superset
         return obj
 
+    def __eq__(self, other):
+        """Return a boolean indicating whether a == b on the basis of
+        whether both objects are of the class Subset and if the values
+        of the subset and superset attributes are the same.
+        """
+        if not isinstance(other, Subset):
+            return NotImplemented
```

Comment:
Don't you want to raise NotImplementedError? Returning as you have it will not cause an error to be raised:
```python
 def ok():
...  return NotImplemented
... 
>>> ok()
NotImplemented
>>>
```

Reply:
https://docs.python.org/3/library/constants.html#NotImplemented

It should tell the interpreter to try the __eq__ of the other object and an error is raised when everything returns `NotImplemented`. 

e.g.
```python
Subset([],[])==0
```
returns `False`, eventhough this would trigger returning `NotImplemented` here.

Is there any specific example you are worried about?

Reply:
No. I didn't understand the mechanics of how NotImplemented works.